### ISSUE TITLE
daemon service does not work in Ubuntu

### ISSUE THREAD

**Issue by loint** [takama/daemon#5]
Hi Igor,

Now it works great in CentOS, but when I change the environment to Ubuntu, it does not work anymore. The service script in "daemon_linux_systemv.go" may be not suitable with ubuntu service pattern. Do you have any idea for this case ?

**Comment by takama** [comment]
Yes, Ubuntu use upstart service manager up to version 16.04 LTS (current 14.04), but will planed changes in service manager in favor of SystemD from version 16.04. Of course, we can develop upstart components also.

**Comment by loint** [comment]
Travis is the most popular CI in Github. But Travis use Ubuntu :(. I think this issue is important with somebody like me who want to do some integration tests with services. Let me give a try with upstart components.

**Comment by takama** [comment]
ok, you are welcome. Let me know when it would be done.
thanks

**Comment by michaeljs1990** [comment]
@loint you have something close to workable? otherwise i'm going to implement this.

**Comment by loint** [comment]
@michaeljs1990 
Currently, I don't have more time to research upstart component so I did with simple trick to make it works with Travis CI in development mode. But it's not a good solution for production. You can reference it at:

https://github.com/penlook/daemon/blob/master/daemon_linux_systemv.go#L312

Now our project is using and it works with both Ubuntu and CentOS.

**Comment by duguying** [comment]
In Ubuntu 14.04 LTS 
```
➜  ~ service elapen stop
/etc/init.d/elapen: 36: /etc/init.d/elapen: [[: not found
```

**Comment by fatihky** [comment]
Hi @duguying you can use my fork currently, I have fixed that issue. Here is my fork: https://github.com/fatihky/daemon

**Comment by onehalfboy** [comment]
ubuntu can use https://github.com/onehalfboy/daemon

**Comment by takama** [comment]
Closed in connection with the pull request #28
